src/pages/CommandBlocklistView/CommandBlocklistView.tsx

```diff
@@ -0,0 +1,200 @@
+import AddIcon from '@mui/icons-material/Add'
+import DeleteIcon from '@mui/icons-material/Delete'
+import { Button, Checkbox, IconButton, Tooltip } from '@mui/material'
+import Divider from 'components/divider'
+import { ModalWrapper } from 'components/ModalWrapper'
+import PageHeader from 'components/PageHeader'
+import { Table } from 'components/Table'
+import { useBlockList } from 'hooks/useBlockList'
+import { generateCommandName, useCommands } from 'hooks/useCommands'
+import { useMemo, useState } from 'react'
+import { Column } from 'react-table'
+import { BlockedCommand, CommandRow } from 'types/custom_types'
+
+const ModalColumns = () => {
+  return useMemo<Column<CommandRow>[]>(
+    () => [
+      {
+        Header: 'Add',
+        accessor: 'action',
+        width: 85,
+      },
+      {
+        Header: 'Namespace',
+        accessor: 'namespace',
+        filter: 'fuzzyText',
+      },
+      {
+        Header: 'System',
+        accessor: 'system',
+        filter: 'fuzzyText',
+      },
+      {
+        Header: 'Command',
+        accessor: 'name',
+      },
+    ],
+    [],
+  )
+}
+
+const TableColumns = () => {
+  return useMemo<Column<CommandRow>[]>(
+    () => [
+      {
+        Header: 'Namespace',
+        accessor: 'namespace',
+        filter: 'fuzzyText',
+      },
+      {
+        Header: 'System',
+        accessor: 'system',
+        filter: 'fuzzyText',
+      },
+      {
+        Header: 'Command',
+        accessor: 'name',
+      },
+      {
+        Header: 'Status',
+        accessor: 'status',
+        filter: 'fuzzyText',
+      },
+      {
+        Header: '',
+        accessor: 'action',
+        width: 85,
+      },
+    ],
+    [],
+  )
+}
+
+export const CommandBlocklistView = () => {
+  const [open, setOpen] = useState(false)
+  const [selectedCommands, setSelected] = useState<CommandRow[]>([])
+  const listClient = useBlockList()
+
+  const CommandlistData = () => {
+    const blocked = useCommands()
+    if (!blocked || !blocked.commands) {
+      return []
+    }
+    return blocked.commands.map((command: CommandRow) => {
+      return commandlistMapper(command)
+    })
+  }
+  const commandlistMapper = (command: CommandRow): CommandRow => {
+    return {
+      namespace: command.namespace,
+      system: command.system,
+      command: command.command,
+      name: command.name,
+      action: (
+        <Checkbox
+          checked={selectedCommands.indexOf(command) > -1}
+          onClick={() => handleClick(command)}
+        />
+      ),
+    }
+  }
+
+  const BlocklistData = () => {
```

Comment:
This should also not be capitalized. It's otherwise fine as is, but a concise refactor might be
```javascript
const blockListData = listClient.getBlockList()?.map(blocklistMapper) ?? []
````

It's not necessary to change it if you prefer the way it is.